data_reader.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+type sourceReader interface {
+	// Unmarshal unmarshals bts to the desired format
+	Unmarshal(bts []byte) (interface{}, error)
+	// IsFile checks whether str contains the file ending
+	IsFile(str string) bool
+}
+
+// readData checks if inputStrSlice provides key=value pairs and validates them using the following input techniques:
+//   * file
+//   * inline
+//   * stdin
+//
+// The "tformat" parameter defines the parser we should use to extract the data.
+func readData(inputStrSlice []string, sr sourceReader) (map[string]interface{}, error) {
+	inputCh := os.Stdin
+	retData := make(map[string]interface{})
+	for _, data := range inputStrSlice {
+		key, value, err := splitVarParam(data)
+		if err != nil {
+			return nil, err
+		}
+
+		byteData := []byte{}
+		if sr.IsFile(value) {
+			byteData, err = ioutil.ReadFile(value)
+			if err != nil {
+				return nil, err
+			}
+		} else if strings.ContainsRune(value, '-') {
```

Comment:
Wieso "contains" ? Das sollte hier doch einfach ein `value=="-"` sein oder nicht?

Reply:
No. In line 32 I already split the data into key and value. After that, I just need to check if the value contains a "-". I have already found a problem with the "contains" thing, which I will fix along with the other changes.